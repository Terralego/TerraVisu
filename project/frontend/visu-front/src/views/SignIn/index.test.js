import React from 'react';
import renderer from 'react-test-renderer';
import { SignInView } from '.';

jest.mock('@terralego/core/modules/Auth', () => ({
  LoginForm: () => <div>LoginForm Mock</div>,
  connectAuthProvider: () => () => null,
}));

describe('rendering', () => {
  it('should render correctly when auth is false', () => {
    const tree = renderer.create((
      <SignInView authenticated={false} logoutAction={() => null} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when auth is true', () => {
    const tree = renderer.create((
      <SignInView authenticated logoutAction={() => null} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
