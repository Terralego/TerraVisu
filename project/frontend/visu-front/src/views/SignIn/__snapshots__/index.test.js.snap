// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render correctly when auth is false 1`] = `
<div>
  LoginForm Mock
</div>
`;

exports[`rendering should render correctly when auth is true 1`] = `
Array [
  <p>
    Vous êtes connecté !
  </p>,
  <button
    onClick={[Function]}
    type="button"
  >
    Déconnexion
  </button>,
]
`;
